src/plugins/query_enhancements/common/utils.ts

```diff
@@ -44,7 +44,23 @@ export const removeKeyword = (queryString: string | undefined) => {
 };
 
 export const throwFacetError = (response: any) => {
-  const error = new Error(response.data.body?.message ?? response.data.body ?? response.data);
+  let errorMessage = response.data.body?.message ?? response.data.body ?? response.data;
+
+  // Check if errorMessage is an object and handle Error objects
+  if (typeof errorMessage === 'object') {
+    if (errorMessage instanceof Error) {
+      // If errorMessage is an instance of Error, extract its message
+      errorMessage = errorMessage.message;
+    } else if (errorMessage.message) {
+      // If errorMessage has a message property, extract that message
+      errorMessage = JSON.stringify(errorMessage.message);
+    } else {
+      // If errorMessage is a plain object, stringify it
+      errorMessage = JSON.stringify(errorMessage);
+    }
+  }
```

Comment:
not for this PR, but looks like there's some more digging we can do to figure out how we can have some sort of response interface here. Right now it looks we're in the wild west 🤠